service/share/discovery/set.go

```diff
@@ -0,0 +1,69 @@
+package discovery
```

Comment:
Pls add note with why this is copy/pasted and link to issue in libp2p

Reply:
Please also add test files / unit tests for this structure

Reply:
I think NOTE is not relevant now bc `peer.Set` was removed at all.

Reply:
Did they remove it already? Damn.

Reply:
yeah